docs/en/operations/settings/settings.md

```diff
@@ -1689,6 +1689,18 @@ Possible values:
 
 Default value: `throw`.
 
+## query_cache_system_table_handling {#query-cache-system-table-handling}
+
+Controls how the [query cache](../query-cache.md) handles `SELECT` queries against system tables, i.e. tables in databases `system.*` and `information_schema.*`.
+
+Possible values:
+
+- `'throw'` - Throw an exception and don't cache the query result.
+- `'save'` - Cache the query result.
+- `'ignore'` - Don't cache the query result and don't throw an exception.
+
+Default value: `throw`.
```

Comment:
btw why default value is `throw`? this seems to be backward incompatible at least, may be better `ignore`?

Reply:
Yes, `throw` is backward incompatible (for an edge case) but it seems the most natural default setting.

Imagine this:
- By default, the query cache is disabled. To use it, one must run a query like this: `SELECT [...] SETTINGS use_query_cache = 1`.
- Users who do this expect that the query result is actually cached (because they explicitly enabled the query cache).
- With setting `ignore`, the query will be executed but the result will  *silently* not be cached. As a user, I would be surprised.

I guess if the query cache was on by default, then setting `ignore` would be okay.

So let's go for the principle of least surprise. `throw` is also consistent with the behavior that happens if the query contains non-deterministic functions (setting `query_cache_nondeterministic_function_handling`).

Reply:
Ok. It just seemed that many users might have query cache enabled by default for all queries. For them it might become a surprise that some queries will start to throw errors. Or someone will enable it by default in the future - same, e.g. he'll have to find out that he'll also need to change another setting. 

Your reasoning also makes sense and I do not have a strong opinion here on what is better, so agreeing with yours.